interfaces/Portal/src/app/services/past-payments.service.ts

```diff
@@ -110,6 +112,7 @@ export class PastPaymentsService {
         person.referenceId,
         transactions,
       );
+      transaction.fspName = this.translatableString.get(transaction.fspName);
```

Comment:
Looking at lines 120-125+ it seems that `transaction` can also be `null`/`undefined`, so this line would mess with that..

So I think moving this one line to inside the `else`, would be safer...

Reply:
We could also look into the 'components that display this' layer, to apply the `translatable-string-magic`....

But for now this Service seems ok.

Reply:
Just move that line to else block, OK, Thanks!